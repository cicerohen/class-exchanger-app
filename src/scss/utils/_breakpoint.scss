@function lower-bound($range) {
  @if length($range) <= 0 {
    @return 0;
  }
  @return nth($range,1);
}

@function upper-bound($range) {
  @if length($range) < 2 {
    @return 999999999999;
  }
  @return nth($range, 2);
}

$small-range:(0em, 39.9375em);
$medium-range:(40em, 63.9375em);
$large-range:(64em, 74.9375em);

$screen: "only screen" !default;
$small-only: "#{$screen} and (max-width: #{upper-bound($small-range)})" !default;
$medium-only: "#{$screen} and (min-width: #{lower-bound($medium-range)}) and (max-width: #{upper-bound($medium-range)})" !default;
$medium-up: "#{$screen} and (min-width: #{lower-bound($medium-range)})" !default;
$large-only: "#{$screen} and (min-width: #{lower-bound($large-range)}) and (max-width: #{upper-bound($large-range)})" !default;
$large-up: "#{$screen} and (min-width: #{lower-bound($large-range)})" !default;
