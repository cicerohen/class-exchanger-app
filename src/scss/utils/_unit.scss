@function strip-unit($num) {
  @return $num / ($num * 0 + 1);
}

@function rem-calc($value, $base: null) {
    
  @if $base == null {
    $base: 62.5%;
  }

  @if unit($base) == '%' {
    $base: ($base / 100%) * 16px;
  }

  @if type-of($value) != 'number' {
    @warn inspect($value) + ' was passed to rem-calc(), which is not a number.';
    @return $value;
  }

  @if unit($value) != 'rem' {
    $value: strip-unit($value) / strip-unit($base) * 1rem;
  }

  @if $value == 0rem {
    $value: 0;
  }
  
  @return $value;

}
