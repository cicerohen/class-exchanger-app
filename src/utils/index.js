import { Iterable, fromJS } from 'immutable';
import { createBrowserHistory } from 'history';

export const asImmutable = obj => (Iterable.isIterable(obj) ? obj : fromJS(obj));
export const emptyMap = asImmutable({});

export const buildActions = (moduleName, prefix, suffix) => ({
    REQUESTED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}REQUESTED`,
    STARTED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}STARTED`,
    SUCCEEDED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}SUCCEEDED`,
    FAILED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}FAILED`,
    FULFILLED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}FULFILLED`,
});

export const toArrayPath = path => path.split('/').filter(p => p.replace(' ', '').length > 0);

export const getHistory = () => {
    window.browserHistory = window.browserHistory || createBrowserHistory();
    return window.browserHistory;
};

export const isDevEnv = () => {
  const nodeEnv = process.env.NODE_ENV;
  return nodeEnv !== 'production' && nodeEnv !== 'test';
};
