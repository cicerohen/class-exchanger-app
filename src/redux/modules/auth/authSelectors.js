import { emptyMap } from '../../../utils';
import MODULE_NAME from './authConstants';

export const getUser = state => state.getIn([`${MODULE_NAME}`, 'user'], emptyMap);
export const isAuthenticating = state => state.getIn([`${MODULE_NAME}`, 'isAuthenticating'], true);
export const isAuthenticated = state => state.getIn([`${MODULE_NAME}`, 'isAuthenticated'], false);
