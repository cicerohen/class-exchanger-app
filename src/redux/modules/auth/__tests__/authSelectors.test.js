import * as selectors from '../authSelectors';
import { asImmutable } from '../../../../utils';

const initialState = {
  auth: {
    user: {
      user: {
        uid: 1,
        email: 'user@email.com',
      }
    },
    isAuthenticated: false,
    isAuthenticating: true,
  }
};

describe('authSelectors tests', () => {
  it('should get user correctly', (done) => {
    expect(selectors.getUser(asImmutable(initialState)).toJS()).toMatchSnapshot();
    done();
  });
  it('should check if is authenticating correctly', (done) => {
    expect(selectors.isAuthenticating(asImmutable(initialState))).toMatchSnapshot();
    done();
  });
  it('should check if is authenticated correctly', (done) => {
    expect(selectors.isAuthenticated(asImmutable(initialState))).toMatchSnapshot();
    done();
  });
});