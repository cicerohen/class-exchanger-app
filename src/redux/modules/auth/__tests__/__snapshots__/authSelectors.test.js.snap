// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authSelectors tests should check if is authenticated correctly 1`] = `false`;

exports[`authSelectors tests should check if is authenticating correctly 1`] = `true`;

exports[`authSelectors tests should get user correctly 1`] = `
Object {
  "user": Object {
    "email": "user@email.com",
    "uid": 1,
  },
}
`;
